packages/models/src/errors.ts

```diff
@@ -144,16 +163,24 @@ export function makeApiProblemBuilder<T extends string>(
     if (operationalLogMessage) {
       logger.warn(operationalLogMessage);
     }
+
     return match<unknown, Problem>(error)
       .with(P.instanceOf(ApiError<T | CommonErrorCodes>), (error) => {
         const mappedCode = httpMapper(error);
-        const code =
-          mappedCode === HTTP_STATUS_INTERNAL_SERVER_ERROR
-            ? defaultCommonErrorMapper(error.code as CommonErrorCodes)
-            : mappedCode;
+        const isInternalServerError =
+          mappedCode === HTTP_STATUS_INTERNAL_SERVER_ERROR;
+
+        const code = isInternalServerError
+          ? defaultCommonErrorMapper(error.code as CommonErrorCodes)
+          : mappedCode;
```

Comment:
By setting `isInternalServerError` before the common error mapper you are skipping the default common error codes mappings!
First we need to apply the `defaultCommonErrorMapper`, then check if it's 500, otherwise we lose mapping of common errors such as unauthorizedError >> 403 👇 

```suggestion
        const code =
          mappedCode === HTTP_STATUS_INTERNAL_SERVER_ERROR
            ? defaultCommonErrorMapper(error.code as CommonErrorCodes)
            : mappedCode;

       const isInternalServerError =
          code === HTTP_STATUS_INTERNAL_SERVER_ERROR;
```